mmv1/third_party/terraform/acctest/vcr_utils_test.go

```diff
@@ -0,0 +1,379 @@
+			},
+		},
+		"requests using different hosts": {
+			httpRequest: requestDescription{
+				scheme: "https",
+				method: "GET",
+				host:   "example.com",
+				path:   "foobar",
+			},
+			cassetteRequest: requestDescription{
+				scheme: "https",
+				method: "GET",
+				host:   "google.com",
+				path:   "foobar",
+			},
+		},
+		"requests using different paths": {
+			httpRequest: requestDescription{
+				scheme: "https",
+				method: "GET",
+				host:   "example.com",
+				path:   "foobar1",
+			},
+			cassetteRequest: requestDescription{
+				scheme: "https",
+				method: "GET",
+				host:   "example.com",
+				path:   "foobar2",
+			},
+		},
+		"requests with different methods": {
+			httpRequest: requestDescription{
+				scheme: "https",
+				method: "POST",
+				host:   "example.com",
+				path:   "foobar",
+				body:   "{}",
+			},
+			cassetteRequest: requestDescription{
+				scheme: "https",
+				method: "PUT",
+				host:   "example.com",
+				path:   "foobar",
+				body:   "{}",
+			},
+		},
+		"POST requests with different bodies": {
+			httpRequest: requestDescription{
+				scheme: "https",
+				method: "POST",
+				host:   "example.com",
+				path:   "foobar",
+				body:   "{\"field\":\"value is ABCDEFG\"}",
+			},
+			cassetteRequest: requestDescription{
+				scheme: "https",
+				method: "POST",
+				host:   "example.com",
+				path:   "foobar",
+				body:   "{\"field\":\"value is MNLOP\"}",
+			},
+		},
+		"POST requests with matching but re-ordered bodies aren't matching if Content-Type header is not 'application/json'": {
+			httpRequest: requestDescription{
+				scheme: "https",
+				method: "POST",
+				host:   "example.com",
+				path:   "foobar",
+				headers: map[string]string{
+					"Content-Type": "foobar",
+				},
+				body: "{\"field1\":\"value1\",\"field2\":\"value2\"}", // 1 before 2
+			},
+			cassetteRequest: requestDescription{
+				scheme: "https",
+				method: "POST",
+				host:   "example.com",
+				path:   "foobar",
+				headers: map[string]string{
+					"Content-Type": "foobar",
+				},
+				body: "{\"field2\":\"value2\",\"field1\":\"value1\"}", // 2 before 1
+			},
+		},
+	}
+
+	for tn, tc := range cases {
+		t.Run(tn, func(t *testing.T) {
+			// Make matcher
+			ctx := context.Background()
+			req := prepareHttpRequest(tc.httpRequest)
+			cassetteReq := prepareCassetteRequest(tc.cassetteRequest)
+			matcher := acctest.NewVcrMatcherFunc(ctx)
+
+			// Act - use matcher
+			matchDetected := matcher(req, cassetteReq)
+
+			// Assert match
+			if matchDetected {
+				t.Fatalf("expected matcher to not match the requests")
+			}
+		})
+	}
+}
+
+// Currently there is no code to actively force the matcher to overlook differing User-Agent values.
+// It isn't checked at any point in the matcher logic.
+func TestNewVcrMatcherFunc_ignoresDifferentUserAgents(t *testing.T) {
+
+	cases := map[string]struct {
+		httpRequest     requestDescription
+		cassetteRequest requestDescription
+		expectMatch     bool
```

Comment:
I just realised that this is no longer needed, but I'll save pushing a commit to remove it until after the automation completes and/or review happens

Reply:
1959f8b824f0d16ec3aa43c9e25d344c83e2a43b